lib/iris/experimental/ugrid/mesh.py

```diff
@@ -1631,6 +1662,39 @@ def coords(
             :class:`Mesh` that matched the given criteria.
 
         """
+        if location is not None:
+            if isinstance(location, str):
+                _location = [location]
+            elif isinstance(location, Iterable):
+                _location = location
+            else:
+                raise TypeError(
+                    f"Expected location to be string or an Iterable, got {type(location)}"
+                )
+            for loc in _location:
+                if loc not in ["node", "edge", "face"]:
+                    raise ValueError(
+                        f"Expected location to contain only `node`, `edge` or `face`, got `{location}`"
+                    )
+            include_nodes_new = "node" in _location
+            include_edges_new = "edge" in _location
+            include_faces_new = "face" in _location
+            if include_nodes not in [None, include_nodes_new]:
+                msg = f"Value of `include_nodes` ({include_nodes}) is incompatible with `location` ({location})"
```

Comment:
Let's not bother with full-on support for both 'types' of parameter. @stephenworsley and I discussed a `DeprecationWarning`, but given the impending ambition to drop the `experimental` label (#5990) I guess we would be stuck supporting the old parameters for a long time.

It's all still `experimental` right now - could we just drop `include_nodes`, `include_edges`, and `include_faces` abruptly. It's not expected to be popular API anyway. Thoughts @bjlittle @pp-mo @stephenworsley?

Reply:
At the very least I don't think we need to be so careful aligning possible duplicate arguments. `location` is in charge, fall back on the other arguments if present.